server/server.go

```diff
@@ -2,18 +2,30 @@ package server
 
 import (
 	"fmt"
-	"github.com/keratin/authn-server/app"
 	"log"
 	"net/http"
+	"time"
+
+	"github.com/keratin/authn-server/app"
 )
 
 func Server(app *app.App) {
 	if app.Config.PublicPort != 0 {
 		go func() {
 			fmt.Println(fmt.Sprintf("PUBLIC_PORT: %d", app.Config.PublicPort))
-			log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", app.Config.PublicPort), PublicRouter(app)))
+			publicServer := &http.Server{
```

Comment:
These changes are in response to gosec warnings on https://app.deepsource.com/directory/analyzers/go/issues/GO-S2112

Might be worth adding timeout to config?

Reply:
Turning it into configuration seems possible, yes, however I currently struggle to imagine it varying by use case. We may prefer to build feedback into a better tuned default.